tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/base/RotBatch.java

```diff
@@ -0,0 +1,31 @@
+package dk.alexandra.fresco.tools.ot.base;
+
+import dk.alexandra.fresco.framework.util.Pair;
+import dk.alexandra.fresco.framework.util.StrictBitVector;
+import java.util.List;
+
+/**
+ * Complete a batch of 1-out-of-2 random OTs, where the receiver gets to pick
+ * its choice bits.
+ */
+public interface RotBatch {
+
+  /**
+   * Send {@code numMessages} of size {@code messageSize} each, in a batch.
+   *
+   * @param numMessages The number of messages in the batch
+   * @param messageSize The size of the messages
+   * @return The random messages construct for each of the OTs
+   */
+  List<Pair<StrictBitVector, StrictBitVector>> send(int numMessages, int messageSize);
+
+  /**
+   * Receive random messages based on the choice bits in {@code choiceBits}.
+   *
+   * @param choiceBits The bits indicating which messages to learn
+   * @param messageSize The size of all the messages
+   * @return The random messages chosen
+   */
+  List<StrictBitVector> receive(StrictBitVector choiceBits, int messageSize);
```

Comment:
The bit vector in the input should not necessarily be strict?

Reply:
Not necessarily, but the code is written to assume that the amount of bits is divisible by 8. Furthermore, because of a bad design choice the OT extension also needs to be able to access bits in a little endian manner, which is currently not a thing I have chosen to add to the bitvector interface. Furthermore, currently, everything else in the OT extension works on StrictBitVector. It will take a lot of time to change this and make sure things still work, without any clear benefit.